dtfs-central-api/src/types/db-models/utilisation-reports.ts

```diff
@@ -37,6 +37,7 @@ export type UtilisationReport = Prettify<
     /**
      * Details of the user that uploaded the report
      */
-    uploadedBy: UploadedByUserDetails;
+    uploadedBy?: UploadedByUserDetails;
+    // TODO FN-1947 is there a nice way to narrow this type? Tried a lot so far which breaks other functions
```

Comment:
I spent quite a bit of time trying to narrow this type so that `uploadedBy` is only `undefined` if the status is `REPORT_NOT_RECEIVED`. I did manage to do it, but it then starts to throw typescript errors with function return types (specifically in the repo files, but I imagine it would be angry elsewhere as well). I think this happens because excessively narrowing a type works quite well when constructing variables to satisfy the type, but trying to get type inference from other places starts to become quite hard for the compiler (`WithoutId<WithId<UtilisationReport>>` no longer exactly matches `UtilisationReport` according to the compiler, for instance). In this case, as much as I would like to have the narrow type, it is easier to just make `uploadedBy` optional. Any thoughts?